IP allowlisting for serverless (#16910)
cockroachdb/docs: src/current/cockroachcloud/connect-to-a-serverless-cluster.md

## Before you start

- [Create a {{ site.data.products.serverless }} cluster](create-a-serverless-cluster.html).
- [Create a new SQL user](managing-access.html#create-a-sql-user).

## Authorize your network

On creation, a Serverless cluster is open to all traffic as it is created with a `0.0.0.0/0` IP allowlist.

> Cockroach Labs recommends restricting your network to allow access only from specific IP address ranges controlled by your organization, according to the [Principle of Least Privilege (PoLP)](https://en.wikipedia.org/wiki/Principle_of_least_privilege). These might include specific networks for your application servers, hardened administrator access points, and backup-restore pipelines for disaster recovery. Therefore, you should replace the `0.0.0.0/0` allowlist entry with more specific CIDR ranges for legitimate access.

Reviewer: ```suggestion
It is recommended to restrict your network to allow access only from specific IP address ranges controlled by your organization. These might include specific networks for your application deployments, hardened administrator access points, or backup-restore pipelines for disaster recovery. Therefore, if possible you should replace the `0.0.0.0/0` allowlist entry with more specific CIDR ranges for legitimate access.
```

Removing or adding an authorized network on your {{ site.data.products.serverless }} cluster may take up to 30 seconds to take effect.

> {% include cockroachcloud/authorize-your-clusters-networks.md %}

Reviewer 2: This include here seems to duplicate some content in the paragraph above?

Reviewer: +1

I would suggest to transfer all IP allowlisting content to that new page, including the context for Serverless (like 0.0.0.0, 30 seconds etc.), and then just include that at all relevant places as is.

Author: Removed duplciate context.
@abhinavg6 the point of that include, which does not render as a separate page, is to include serverless/dedicated agnostic content that is re-used between the page, so it can't contain serverless stuff.
I'm not sure if you a referring to a different 'separate page', but to be clear this does not render as a separate page to the user (consult the previews linked at the top of the PR to see how it renders in the docs). Sorry if I am misunderstanding your comment here.
